src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -873,6 +873,16 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 }
             }
 
+            // Require a separator between a lambda return type and its open paren.
+            // Current token might be 2 or 3 layers deeper than lambda expression.
+            // If the type syntax is simple, e.g. int () => ... then the token is 2 layers deeper.
+            // In more complex cases, e.g. int[] () => ..., A.B* () => ... etc. it is 3 layers deeper that the lambda.
+            if (token.Parent is { Parent: LambdaExpressionSyntax or { Parent: LambdaExpressionSyntax } } &&
+                next is { RawKind: (int)SyntaxKind.OpenParenToken, Parent.Parent: LambdaExpressionSyntax })
+            {
+                return true;
+            }
```

Comment:
the token.parent check seems suspect to me.    Why not just check if `next` is the start paren.  If so, you have a lambda.  Then just check if `token` is the last token if the return-type of that lambda?  Much simpler.